src/quickgui/plugin/qgsquickpositionmarker.qml

```diff
@@ -0,0 +1,155 @@
+/***************************************************************************
+  qgsquickpositionmarker.qml
+  --------------------------------------
+  Date                 : Dec 2017
+  Copyright            : (C) 2017 by Peter Petrik
+  Email                : zilolv at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick 2.3
+import QtQuick.Controls 2.2
+import QtQml 2.2
+import QtGraphicalEffects 1.0
+import QgisQuick 0.1 as QgsQuick
+
+Item {
+  id: positionMarker
+  property int size: 48 * QgsQuick.Utils.dp
+
+  property QgsQuick.MapSettings mapSettings // required to be connected from parent!
+  property QgsQuick.PositionKit positionKit: QgsQuick.PositionKit {id: positionKit}
+
+  property var simulatePositionLongLatRad // use in debug mode to simulate movement around some GPS location
```

Comment:
Shouldn't that rather go to the position kit than to this marker?

Reply:
You are right. I have a feeling that there is too much logic in this QML file and most of it should be transferred to C++, at least in a longer term. I  will try to address it in the separate PR.